code/game/machinery/Sleeper.dm

```diff
@@ -18,21 +18,45 @@
 	var/min_health = -25
 	var/list/available_chems
 	var/controls_inside = FALSE
-	var/list/possible_chems = list(
+	var/obj/item/reagent_containers/list/chem_stores
+	var/list/chem_store_names
+	var/selected_store = 0
+	var/list/possible_chem_stores = list(3, 7, 12, 20)
+	var/list/possible_chems
+	var/list/mapload_containers = list(/obj/item/reagent_containers/glass/bottle/epinephrine, /obj/item/reagent_containers/glass/bottle/bicaridine, /obj/item/reagent_containers/glass/bottle/kelotane)
+	var/stasis_enabled = TRUE //Stores whether stasis is turned on, isn't intended to disable stasis for a subtype
+	var/list/chem_buttons	//Used when emagged to scramble which chem is used, eg: antitoxin -> morphine
+	var/scrambled_chems = FALSE //Are chem buttons scrambled? used as a warning
+	var/enter_message = "<span class='notice'><b>You feel cool air surround you. You go numb as your senses turn inward.</b></span>"
+
+/obj/machinery/sleeper/chems
+	possible_chems = list(
 		list("epinephrine", "morphine", "salbutamol", "bicaridine", "kelotane"),
 		list("oculine","inacusiate"),
 		list("antitoxin", "mutadone", "mannitol", "pen_acid"),
 		list("omnizine")
 	)
-	var/list/chem_buttons	//Used when emagged to scramble which chem is used, eg: antitoxin -> morphine
-	var/scrambled_chems = FALSE //Are chem buttons scrambled? used as a warning
-	var/enter_message = "<span class='notice'><b>You feel cool air surround you. You go numb as your senses turn inward.</b></span>"
+	possible_chem_stores = null
+
+/obj/machinery/sleeper/empty
+	mapload_containers = null
 
-/obj/machinery/sleeper/Initialize()
+/obj/machinery/sleeper/Initialize(mapload)
 	. = ..()
 	update_icon()
 	reset_chem_buttons()
 
+	if(mapload && mapload_containers && chem_stores)

```

Comment:
This is causing problems with the holodeck sleepers, completely forgot about that

Reply:
This is because I'm moving the containers to nullspace instead of putting them inside the machine, i'll try to do this if I can.